clients/ui/frontend/src/app/pages/modelCatalogSettings/components/ManageSourceForm.tsx

```diff
@@ -0,0 +1,134 @@
+  const navigate = useNavigate();
+  const [formData, setData] = useManageSourceData(existingData);
+  const [isFiltersExpanded, setIsFiltersExpanded] = React.useState(false);
+  const [isSubmitting, setIsSubmitting] = React.useState(false);
+  const [submitError, setSubmitError] = React.useState<Error | undefined>(undefined);
+
+  const isHuggingFaceMode = formData.sourceType === SourceType.HuggingFace;
+  const isFormComplete = isFormValid(formData);
+  const canPreview = isPreviewReady(formData);
+
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    setSubmitError(undefined);
+
+    try {
+      // TODO: Implement submit logic (will be part of API integration)
+      // navigate(catalogSettingsUrl());
+    } catch (error) {
+      setSubmitError(error instanceof Error ? error : new Error('Failed to save source'));
+    } finally {
+      setIsSubmitting(false);
+    }
+  };
+
+  const handlePreview = () => {
+    // TODO: Implement preview logic (will be part of API integration)
+  };
+
+  const handleCancel = () => {
+    navigate(catalogSettingsUrl());
+  };
+
+  return (
+    <>
+      <Sidebar hasGutter hasBorder isPanelRight>
+        <SidebarContent>
+          <Form isWidthLimited>
+            <Stack hasGutter>
+              <StackItem>
+                <SourceDetailsSection formData={formData} setData={setData} />
+              </StackItem>
+
+              {isHuggingFaceMode && (
+                <StackItem>
+                  <CredentialsSection formData={formData} setData={setData} />
+                </StackItem>
+              )}
+
+              {!isHuggingFaceMode && (
+                <StackItem>
+                  <YamlSection formData={formData} setData={setData} />
+                </StackItem>
+              )}
+
+              <StackItem>
+                <ModelVisibilitySection
+                  formData={formData}
+                  isExpanded={isFiltersExpanded}
+                  onToggle={() => setIsFiltersExpanded((prev) => !prev)}
+                  setData={setData}
+                />
+              </StackItem>
+
+              <StackItem>
+                <FormSection>
+                  <FormGroup fieldId="enable-source">
+                    <Checkbox
+                      label={<strong>{FORM_LABELS.ENABLE_SOURCE}</strong>}
+                      id="enable-source"
+                      name="enable-source"
+                      data-testid="enable-source-checkbox"
+                      description={DESCRIPTIONS.ENABLE_SOURCE}
+                      isChecked={formData.enabled}
+                      onChange={(_event, checked) => setData('enabled', checked)}
+                    />
+                  </FormGroup>
+                </FormSection>
+              </StackItem>
+            </Stack>
+          </Form>
+        </SidebarContent>
+        <SidebarPanel hasPadding className="pf-v6-u-flex-basis-40">
```

Comment:
All you need here is the `width` prop on SidebarPanel. [You can set different widths by screen size](https://www.patternfly.org/components/sidebar#responsive-panel-width), but I think we're fine just having it always be 50%.

```suggestion
        <SidebarPanel hasPadding width={{ default: 'width_50' }}>
```